src/USER-MISC/angle_fourier_simple_approx.cpp

```diff
@@ -0,0 +1,218 @@
+// clang-format off
+/* ----------------------------------------------------------------------
+   LAMMPS - Large-scale Atomic/Molecular Massively Parallel Simulator
+   https://www.lammps.org/, Sandia National Laboratories
+   Steve Plimpton, sjplimp@sandia.gov
+
+   Copyright (2003) Sandia Corporation.  Under the terms of Contract
+   DE-AC04-94AL85000 with Sandia Corporation, the U.S. Government retains
+   certain rights in this software.  This software is distributed under
+   the GNU General Public License.
+
+   See the README file in the top-level LAMMPS directory.
+------------------------------------------------------------------------- */
+
+/* ----------------------------------------------------------------------
+   Contributing author: Tim Bernhard (ETHZ)
+------------------------------------------------------------------------- */
+
+#include "angle_fourier_simple_approx.h"
+#include <cmath>
+#include "atom.h"
+#include "comm.h"
+#include "force.h"
+#include "neighbor.h"
+
+
+#include "suffix.h"
+using namespace LAMMPS_NS;
+
+#define SMALL 0.001
+#define TWO_PI 2.*M_PI
+
+
+typedef struct { double x,y,z; } dbl3_t;
+typedef struct { int a,b,c,t;  } int4_t;
+
+/* ---------------------------------------------------------------------- */
+
+AngleFourierSimpleApprox::AngleFourierSimpleApprox(class LAMMPS *lmp)
+  : AngleFourierSimple(lmp)
+{
+  suffix_flag |= Suffix::OMP;
+}
+
+/* ---------------------------------------------------------------------- */
+
+/**
+ * Approximation of cos
+ * maximum error is about 0.00109 for the range -pi to pi
+ * Source: https://stackoverflow.com/a/28050328/3909202
+ */
+double AngleFourierSimpleApprox::fastCos(double x) {
+  constexpr double tp = 1./(TWO_PI);
+  // TODO: check if range map is necessary
+  double x_wrapped = x - TWO_PI * floor(x * tp);
+    x_wrapped *= tp;
+    x_wrapped -= .25 + std::floor(x_wrapped + .25);
+    x_wrapped *= 16. * (std::abs(x_wrapped) - .5);
+    x_wrapped += .225 * x_wrapped * (std::abs(x_wrapped) - 1.);
+    return x_wrapped;
+}
+
+/* ---------------------------------------------------------------------- */
+
+/**
+ * Approximation of acos 
+ * Returns the arccosine of x in the range [0,pi], expecting x to be in the range [-1,+1]. 
+ * Absolute error <= 6.7e-5
+ * Range mapping not necessary as C++ standard library would throw domain error
+ * Source: https://developer.download.nvidia.com/cg/acos.html
+ */
+double AngleFourierSimpleApprox::fastAcos(double x) {
+  double negate = double(x < 0);
+  x = abs(x);
```

Comment:
Is this supposed to be `abs()` (operates on and returns an integer) or `fabs()` (operates on and returns a double precision floating point number)?